firmware/Core/Src/telecommands/i2c_telecommand_defs.c

```diff
@@ -0,0 +1,68 @@
+#include "i2c_telecommand_defs.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "main.h"
+
+uint8_t TCMDEXEC_scan_i2c_bus(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                        char *response_output_buf, uint16_t response_output_buf_len) {
+
+    I2C_HandleTypeDef* hi2c;
+    uint8_t bus_to_scan = atoi(args_str);
+
+    // Figure out which bus we want to scan
+    switch(bus_to_scan){
+        case 1:
+            hi2c = &hi2c1;
+            break;
+        case 2:
+            hi2c = &hi2c2;
+            break;
+        case 3: 
+            hi2c = &hi2c3;
+            break;
+        case 4:
+            hi2c = &hi2c4;
+            break;
+        default:
+            hi2c = NULL;
+            snprintf(response_output_buf, response_output_buf_len, "Enter a valid I2C bus!\n");
+            return 0;
+    }
+
+	char msg[6];
+    HAL_StatusTypeDef i2c_device_status;
+
+	// Go through all possible i2c addresses
+	for (uint8_t i = 0; i < 128; i++) {
+
+        i2c_device_status = HAL_I2C_IsDeviceReady(hi2c, (uint16_t)(i<<1), 3, 5);
+
+		if (i2c_device_status == HAL_OK) {
+			// O for OK
+			sprintf(msg, "%2x O\n", i);
```

Comment:
Please use `snprintf` for length overflow safety. No `snprintf`. `strcat` is also unsafe.

Reply:
Done.